### PR TITLE
OSD-15763 & OSD-15766 : Fix the program crashing error when many slides are removed at once

### PR DESCRIPTION
**Pull request by Dee-6777**
### What type of PR is this?

_bug_

### What this PR does / Why we need it?
Solves the bug which is occurred when any number of pages are added or deleted and the app crashes.
Also solves app crashing when exit command & remove slide are executed simultaneously.
### Which Jira/Github issue(s) does this PR fix?

- _Solves_ #[OSD-15766](https://issues.redhat.com/browse/OSD-15766)
- _Solves_ #[OSD-15763](https://issues.redhat.com/browse/OSD-15763)
### Pre-checks (if applicable)

- [ ] Ran unit tests locally against the changes
- [ ] Included documentation changes with PR

### CODE REVIEW

**Review by devppratik — CHANGES_REQUESTED** [review]

File: pkg/ui/tab.go [hunk 1]
```diff
@@ -62,21 +62,21 @@ func newTabPrimitive(command string, args []string) (content tview.Primitive) {
...
-	tui.TerminalPageBar.Highlight(strconv.Itoa(tui.TerminalUIRegionIDs[CurrentActivePage])).
+	CurrentActivePage = (CurrentActivePage + 1 + len(tui.TerminalTabs)) % len(tui.TerminalTabs)
+	tui.TerminalPageBar.Highlight(strconv.Itoa(tui.TerminalTabs[CurrentActivePage].index)).
 		ScrollToHighlight()
 	tui.TerminalInputBuffer = []rune{}
 }
 
-func indexOf(arr []int, ele int) int {
-	for index, item := range arr {
+func indexOf(RegionIDs []int, ele int) int {
```
[the remark this answers is not in the document]
Comment (Dee-6777) [reply pkg/ui/tab.go:78]: Thanks for the suggestion! Added it.

File: pkg/ui/tab.go [hunk 2]
```diff
@@ -97,26 +97,29 @@ func RemoveSlide(s int, tui *TUI) {
...
 		}
 	}
 	tabSlide := NewTab(name, command, args, tui)
-	tui.TerminalTabs = append(tui.TerminalTabs, *tabSlide)
-	tui.TerminalPages.AddPage(strconv.Itoa(tabSlide.index), tabSlide.primitive, true, tabSlide.index == 0)
-	fmt.Fprintf(tui.TerminalPageBar, `["%d"]%s[white][""]  `, tabSlide.index, fmt.Sprintf("%d %s", tabSlide.index+1, tabSlide.title))
-	CurrentActivePage = tabSlide.index
-	tui.TerminalPageBar.Highlight(strconv.Itoa(CurrentActivePage)).
-		ScrollToHighlight()
+	if len(tui.TerminalTabs) < 9 {
```
[the remark this answers is not in the document]
Comment (Dee-6777) [reply pkg/ui/tab.go:115]: Yes sure!

File: pkg/ui/tab.go [hunk 3]
```diff
@@ -97,26 +97,29 @@ func RemoveSlide(s int, tui *TUI) {
...
 	PreviousSlide(tui)
+	tui.TerminalInputBuffer = []rune{}
 }
 
 // Adds a slide to the end of currently present slides
 func AddNewSlide(tui *TUI, name string, command string, args []string, isCluster bool) {
 	if isCluster {
 		for i, tab := range tui.TerminalTabs {
-			if tab.title == args[0] {
+			if tab.primitive != nil {
```
[the remark this answers is not in the document]
Comment (Dee-6777) [reply pkg/ui/tab.go:107]: Corrected. Thanks for highlighting this!

**Review by supreeth7 — APPROVED** [review]
/lgtm

**Pull request merged**